### PR TITLE
More Meteor defenses

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
[Changelogs]
Makes a new meteor defenses, Station made meteor shields are COSTLY and not worth making when you get safer and less dangers stats on cargo.
A new disk that reduces the odds of meteors per stat as well as extends its range for 3 more tiles 
MISC organizations so its better to read at a eyes glance 
Play Tested!
image
[why]
Misc needed to be better organized
Meteor shields are allmost never gotten, do to engi lazyness or cargo not wanting to deal with it
This hopefully will allow for engis to set 1-3 of them around high risk meteor areas, or quickly throw one up by the SM if needed. Meteor shields are legit used less then any other station goal, but increases the odds for meteors. So spam buying them and letting them sit in cargo makes meteors happen MORE offten - Or I cant read code...

### CODE REVIEW

File: code/modules/station_goals/shield.dm [hunk 1]
```diff
@@ -134,6 +134,32 @@
...
+
+/obj/item/disk/meteor
+	name = "Meteor Shield Upgrade Disk"
+	desc = "A floppy disk that allows meteor shields to fire at longer ranges and lowers meteor drawing from gravitational fields.."
+
+/obj/machinery/satellite/meteor_shield/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/disk/meteor))
+		to_chat(user, "<span class='notice'>The disk uploads better tracking and rang modification software.</span>")
+		kill_range = 17
+		change_meteor_chance(0.5)
```
Comment (deathride58) [inline code/modules/station_goals/shield.dm]: This makes it plausible to completely nullify the meteor chance by repeatedly spam-clicking a single meteor shield with an upgrade disk. Make this check to see if the disk has already been applied

File: code/modules/station_goals/shield.dm [hunk 2]
```diff
@@ -134,6 +134,32 @@
...
+		return ..()
+
+/obj/machinery/satellite/meteor_shield/sci/toggle(user)
+	if(!..(user))
+		return FALSE
+	if(obj_flags & EMAGGED)
+		if(active)
+			change_meteor_chance(8)
+		else
+			change_meteor_chance(1.5)
```
Comment (deathride58) [inline code/modules/station_goals/shield.dm]: This makes it plausible to guarantee meteors by emagging a meteor shield and repeatedly toggling it on/off. Make this 0.125

**Review by deathride58 — CHANGES_REQUESTED** [review]

File: code/modules/research/techweb/all_nodes.dm [hunk 3]
```diff
@@ -203,6 +203,25 @@
 	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 3000)
 	export_price = 5000
 
+/datum/techweb_node/basic_meteor_defense
+	id = "basic_meteor_defense"
+	display_name = "Meteor Defense Research"
+	description = "Unlock the potential of the mysterious of why CC desided to not build these around the staition themselfs."
```
Comment (silicons) [inline code/modules/research/techweb/all_nodes.dm]: fix this

File: code/modules/station_goals/shield.dm [hunk 4]
```diff
@@ -134,6 +134,31 @@
...
+	desc = "A station made meteor point-defense satellite."
+	mode = "M-SHIELD"
+
+/obj/item/disk/meteor
+	name = "Meteor Shield Upgrade Disk"
+	desc = "A floppy disk that allows meteor shields to fire at longer ranges and lowers meteor drawing from gravitational fields.."
+
+/obj/machinery/satellite/meteor_shield/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/disk/meteor))
+		to_chat(user, "<span class='notice'>The disk uploads better tracking and rang modification software.</span>")
```
Comment (silicons) [inline code/modules/station_goals/shield.dm:148]: focusing, not range modification, or just remove that part

File: code/modules/station_goals/shield.dm [hunk 5]
```diff
@@ -134,6 +134,31 @@
...
+		kill_range = 17
+	else
+		return ..()
+
+/obj/machinery/satellite/meteor_shield/sci/toggle(user)
+	if(!..(user))
+		return FALSE
+	if(obj_flags & EMAGGED)
+		if(active)
+			change_meteor_chance(8)
```
Comment (silicons) [inline code/modules/station_goals/shield.dm:158]: uh where are these values from

File: code/modules/research/designs/misc_designs.dm [hunk 6]
```diff
@@ -469,3 +474,28 @@
...
+/////////////////////////////////////////
+////////////Meteor///////////////////////
+/////////////////////////////////////////
+
+/datum/design/meteor_defence
+	name = "Meteor Defence"
+	desc = "A blue print of a early model of the Meteor defence turret."
+	id = "meteor_defence"
+	build_type = PROTOLATHE
+	materials = list(MAT_METAL = 150000, MAT_GLASS = 150000, MAT_SILVER = 7500, MAT_GOLD = 10000, MAT_TITANIUM = 35000) //Station goal
```
Comment (silicons) [inline code/modules/research/designs/misc_designs.dm]: Making things cost a shitload of a certain type of materials is not how you should be balancing these things. Make them cost more material types, but less metal/glass, why does a single shield need as much metal as to build a small room?

**Review by silicons — CHANGES_REQUESTED** [review]

File: code/modules/research/designs/misc_designs.dm (hunk 6 above)
Comment (Trilbyspaceclone) [reply code/modules/research/designs/misc_designs.dm]: So we dont spam them around ever shift, that way cargo is main force behind buying them. I did the math and its around 20 metal and glass when maxed well others are less then 12~ each

File: code/modules/station_goals/shield.dm (hunk 5 above)
Comment (Trilbyspaceclone) [reply code/modules/station_goals/shield.dm:158]: Helps shift up meteor odds, thats if Emagged and active gives +8% odds of meteor happening if im reading the code correctly

File: code/modules/research/techweb/all_nodes.dm (hunk 3 above)
Comment (Trilbyspaceclone) [reply code/modules/research/techweb/all_nodes.dm]: Fix what?

File: code/modules/station_goals/shield.dm (hunk 4 above)
Comment (Trilbyspaceclone) [reply code/modules/station_goals/shield.dm:148]: This only improves the range by 3 tiles, nothing more. This IC feedback is 100% vaild

File: code/modules/station_goals/shield.dm [hunk 7]
```diff
@@ -134,6 +134,31 @@
...
+		else
+			change_meteor_chance(1.25)
```
Comment (deathride58) [inline code/modules/station_goals/shield.dm]: Still needs to be changed to 0.125, change_meteor_chance is multiplicative

**Review by deathride58 — DISMISSED** [review]

File: code/modules/station_goals/shield.dm (hunk 5 above)
Comment (deathride58) [reply code/modules/station_goals/shield.dm:158]: This line of code multiplies the meteor chance 8x

**Comment by silicons** [comment]
@Citadel-Station-13/project-management how do we proceed i don't like this but i also don't dislike this enough to just close it, it's fundamentally flawed in balance atm imo but i don't know what to do about it.

**Review by silicons — APPROVED** [review]

**Pull request merged**